Source/Particles/Collision/BinaryCollision/DSMC/SplitAndScatterFunc.H

```diff
@@ -0,0 +1,168 @@
+            // If the colliding particle weight decreases to zero, remove particle by
+            // setting its id to -1.
+            Gpu::Atomic::AddNoRet(&w1, -p_pair_reaction_weight[i]);
+            if (w1 <= 0._prt) {
+                auto& p = ptile1_data.m_aos[p_pair_indices_1[i]];
+                p.atomicSetID(minus_one_long);
+            }
+
+            Gpu::Atomic::AddNoRet(&w2, -p_pair_reaction_weight[i]);
+            if (w2 <= 0._prt) {
+                auto& p = ptile2_data.m_aos[p_pair_indices_2[i]];
+                p.atomicSetID(minus_one_long);
+            }
+
+            // Set the child particle properties appropriately
+            ptile1_data.m_rdata[PIdx::w][p_offsets[i] + ptile1_index] = p_pair_reaction_weight[i];
+            ptile2_data.m_rdata[PIdx::w][p_offsets[i] + ptile2_index] = p_pair_reaction_weight[i];
+
+            auto& ux1 = ptile1_data.m_rdata[PIdx::ux][p_offsets[i] + ptile1_index];
+            auto& uy1 = ptile1_data.m_rdata[PIdx::uy][p_offsets[i] + ptile1_index];
+            auto& uz1 = ptile1_data.m_rdata[PIdx::uz][p_offsets[i] + ptile1_index];
+            auto& ux2 = ptile2_data.m_rdata[PIdx::ux][p_offsets[i] + ptile2_index];
+            auto& uy2 = ptile2_data.m_rdata[PIdx::uy][p_offsets[i] + ptile2_index];
+            auto& uz2 = ptile2_data.m_rdata[PIdx::uz][p_offsets[i] + ptile2_index];
+
+            // for simplicity (for now) we assume non-relativistic particles
+            // and simply calculate the center-of-momentum velocity from the
+            // rest masses
+            auto const uCOM_x = (m1 * ux1 + m2 * ux2) / (m1 + m2);
+            auto const uCOM_y = (m1 * uy1 + m2 * uy2) / (m1 + m2);
+            auto const uCOM_z = (m1 * uz1 + m2 * uz2) / (m1 + m2);
+
+            // transform to COM frame
+            ux1 -= uCOM_x;
+            uy1 -= uCOM_y;
+            uz1 -= uCOM_z;
+            ux2 -= uCOM_x;
+            uy2 -= uCOM_y;
+            uz2 -= uCOM_z;
+
+            if (mask[i] == int(ScatteringProcessType::ELASTIC)) {
+                // randomly rotate the velocity vector for the first particle
+                ParticleUtils::RandomizeVelocity(
+                    ux1, uy1, uz1, std::sqrt(ux1*ux1 + uy1*uy1 + uz1*uz1), engine
+                );
+                // set the second particles velocity so that the total momentum
+                // is zero
+                ux2 = -ux1 * m1 / m2;
+                uy2 = -uy1 * m1 / m2;
+                uz2 = -uz1 * m1 / m2;
+            } else if (mask[i] == int(ScatteringProcessType::BACK)) {
+                // reverse the velocity vectors of both particles
+                ux1 *= -1.0_prt;
+                uy1 *= -1.0_prt;
+                uz1 *= -1.0_prt;
+                ux2 *= -1.0_prt;
+                uy2 *= -1.0_prt;
+                uz2 *= -1.0_prt;
+            } else if (mask[i] == int(ScatteringProcessType::CHARGE_EXCHANGE)) {
```

Comment:
In the future, for charge exchange through binary collisions, maybe we could replace this implementation by a separate one that works directly with the ionization level array (which is used already for field ionization).
This would allow field ionization, charge exchange and (later) impact ionization to all work within the same framework.

Reply:
That would be awesome!